### PR TITLE
Add support to compress data in PrestoVectorSerializer

### PR DESCRIPTION
**Pull request by jinchengchenghh**
Compress the data by folly when flush to OutputStream. And decompress the data under deserialization.
Will use in spill compression.

Resolve:https://github.com/facebookincubator/velox/issues/5313

### CODE REVIEW

File: velox/serializers/PrestoSerializer.cpp [hunk 1]
```diff
@@ -1595,9 +1595,12 @@ class PrestoVectorSerializer : public VectorSerializer {
       std::shared_ptr<const RowType> rowType,
       int32_t numRows,
       StreamArena* streamArena,
-      bool useLosslessTimestamp) {
+      bool useLosslessTimestamp,
+      dwio::common::CompressionKind compressionKind) {
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1599]: Can make streamArena_ and codec_ const members of PrestoVectorSerializer? Thanks!

File: velox/serializers/PrestoSerializer.cpp [hunk 2]
```diff
@@ -1699,12 +1698,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+      flushCompressed(numRows, rle, out, listener);
+    }
+  }
+
  private:
   static const int32_t kSizeInBytesOffset{4 + 1};
   static const int32_t kHeaderSize{kSizeInBytesOffset + 4 + 4 + 8};
 
   int32_t numRows_{0};
+  StreamArena* streamArena_;
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1785]:
```
StreamArena* const streamArena_;
const std::unique_ptr<folly::io::Codec> codec_;
```
And put const members first?

File: velox/serializers/PrestoSerializer.cpp [hunk 3]
```diff
@@ -1699,12 +1698,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+
+  // Writes the contents to 'stream' in wire format
+  void flushInternal(int32_t numRows, bool rle, OutputStream* out) {
+    auto listener = dynamic_cast<PrestoOutputStreamListener*>(out->listener());
+    // Reset CRC computation
+    if (listener) {
+      listener->reset();
+    }
+
+    if (codec_->type() == folly::io::CodecType::NO_COMPRESSION) {
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1773]: Consider to add a utility
```
FOLLY_ALWAYS_INLINE bool needCompression() const;
```

File: velox/serializers/PrestoSerializer.cpp [hunk 4]
```diff
@@ -1720,11 +1800,22 @@ std::unique_ptr<VectorSerializer> PrestoVectorSerde::createSerializer(
     int32_t numRows,
     StreamArena* streamArena,
     const Options* options) {
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1802]: Does it require a lot of changes if we make options a const &? and provide a default one on ctor if not specified?

File: velox/serializers/PrestoSerializer.cpp [hunk 5]
```diff
@@ -1747,6 +1838,11 @@ void PrestoVectorSerde::deserialize(
   bool useLosslessTimestamp = options != nullptr
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1838]:
```
NYC: const bool useLosslessTimestamp =
```

File: velox/serializers/PrestoSerializer.cpp [hunk 6]
```diff
@@ -1747,6 +1838,11 @@ void PrestoVectorSerde::deserialize(
...
       ? static_cast<const PrestoOptions*>(options)->useLosslessTimestamp
       : false;
+  std::unique_ptr<folly::io::Codec> codec = options == nullptr
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1841]: nit: we can have an anonymous function which takes options ptr to return the codec
```
std::unique_ptr<folly::io::Codec> getCompressionCodec(options* );
```

File: velox/serializers/PrestoSerializer.cpp [hunk 7]
```diff
@@ -1720,11 +1800,22 @@ std::unique_ptr<VectorSerializer> PrestoVectorSerde::createSerializer(
...
-      type, numRows, streamArena, useLosslessTimestamp);
+  if (options == nullptr) {
+    return std::make_unique<PrestoVectorSerializer>(
+        type,
+        numRows,
+        streamArena,
+        false,
+        dwio::common::CompressionKind::CompressionKind_NONE);
+  } else {
+    auto prestoOptions = static_cast<const PrestoOptions*>(options);
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1811]: nit: provide a utility: toPrestoOptions() for this conversion? Thanks!

File: velox/serializers/PrestoSerializer.cpp [hunk 8]
```diff
@@ -1758,24 +1854,40 @@ void PrestoVectorSerde::deserialize(
   auto pageCodecMarker = source->read<int8_t>();
   auto uncompressedSize = source->read<int32_t>();
   // skip size in bytes
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1856]: Update the comment here?

File: velox/serializers/PrestoSerializer.cpp [hunk 9]
```diff
@@ -1758,24 +1854,40 @@ void PrestoVectorSerde::deserialize(
...
       checksum, actualCheckSum, "Received corrupted serialized page.");
 
-  // skip number of columns
-  source->skip(4);
-
   auto children = &(*result)->children();
   auto childTypes = type->as<TypeKind::ROW>().children();
-  readColumns(source, pool, childTypes, children, useLosslessTimestamp);
+  if (codec->type() == folly::io::CodecType::NO_COMPRESSION) {
+    VELOX_CHECK(!isCompressedBitSet(pageCodecMarker));
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1872]: Shall we move the check above at l1845?
```
VELOX_CHECK_EQ(codec->type() == folly::io::CodecType::NO_COMPRESSION, !isCompressedBitSet(pageCodecMarker), "error message");
```

File: velox/serializers/PrestoSerializer.cpp [hunk 10]
```diff
@@ -1699,12 +1698,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+      writeInt32(&out, kRLE.size());
+      out.write(kRLE.data(), kRLE.size());
+      // Write number of RLE values.
+      writeInt32(&out, numRows);
+    }
+
+    for (auto& stream : streams_) {
+      stream->flush(&out);
+    }
+    int32_t uncompressedSize = out.tellp();
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1733]: const

File: velox/serializers/PrestoSerializer.cpp [hunk 11]
```diff
@@ -1699,12 +1698,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+    VELOX_CHECK_LE(
+        uncompressedSize,
+        codec_->maxUncompressedLength(),
+        "UncompressedSize exceeds limit");
+    auto compressed = codec_->compress(out.getIOBuf().get());
+    int32_t compressedSize = compressed->length();
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1739]: nit: const

File: velox/serializers/PrestoSerializer.cpp [hunk 12]
```diff
@@ -1699,12 +1698,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+    if (listener) {
+      listener->pause();
+    }
+    // Fill in crc
+    int64_t crc = 0;
+    if (listener) {
+      crc = computeChecksum(listener, codec, numRows, compressedSize);
+    }
+    output->seekp(crcOffset);
+    writeInt64(output, crc); // Write zero checksum
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1761]: The comment is not valid?

File: velox/serializers/tests/PrestoSerializerTest.cpp [hunk 13]
```diff
@@ -147,6 +149,18 @@ TEST_F(PrestoSerializerTest, basic) {
   testRoundTrip(rowVector);
 }
 
+TEST_F(PrestoSerializerTest, compress) {
```
Comment (xiaoxmeng) [inline velox/serializers/tests/PrestoSerializerTest.cpp:152]: Can you parameterize this test file to cover each test case with two different modes: compressed and uncompressed? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
Thanks for the change % minors!

File: velox/serializers/PrestoSerializer.cpp (hunk 4 above)
Comment (jinchengchenghh) [reply velox/serializers/PrestoSerializer.cpp:1802]: This will be a lot of change and non-relevant change, we can do that in another PR if needed.

File: velox/serializers/PrestoSerializer.cpp (hunk 1 above)
Comment (jinchengchenghh) [reply velox/serializers/PrestoSerializer.cpp:1599]: If make codec_ const, I will receive error:
```
no operator "=" matches these operandsC/C++(349)
PrestoSerializer.cpp(1617, 12): operand types are: const std::unique_ptr<folly::io::Codec, std::default_delete<folly::io::Codec>> = std::unique_ptr<folly::io::Codec, std::default_delete<folly::io::Codec>>
```

**Comment by jinchengchenghh** [comment]
Fixed all the comments, can you help review again? @xiaoxmeng

File: velox/serializers/PrestoSerializer.h [hunk 14]
```diff
@@ -15,20 +15,29 @@
...
 namespace facebook::velox::serializer::presto {
 class PrestoVectorSerde : public VectorSerde {
  public:
   // Input options that the serializer recognizes.
   struct PrestoOptions : VectorSerde::Options {
-    explicit PrestoOptions(bool useLosslessTimestamp)
-        : useLosslessTimestamp(useLosslessTimestamp) {}
+    explicit PrestoOptions() {}
+
+    explicit PrestoOptions(
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.h:28]: Drop explicit which is only required for single input ctor.

File: velox/serializers/PrestoSerializer.h [hunk 15]
```diff
@@ -15,20 +15,29 @@
...
+        bool useLosslessTimestamp,
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.h:29]:
```
s/useLosslessTimestamp/_useLosslessTimestamp/
s/compressionKind/_compressionKind/
```

File: velox/serializers/PrestoSerializer.cpp [hunk 16]
```diff
@@ -136,6 +136,18 @@ std::string typeToEncodingName(const TypePtr& type) {
   }
 }
 
+const PrestoVectorSerde::PrestoOptions toPrestoOptions(
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:139]:
```
s/const PrestoVectorSerde::PrestoOptions/PrestoVectorSerde::PrestoOptions/
```
Drop the const as return by value?

File: velox/serializers/PrestoSerializer.cpp [hunk 17]
```diff
@@ -1699,10 +1710,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+    } else {
+      flushCompressed(numRows, rle, out, listener);
+    }
+  }
+
  private:
   static const int32_t kSizeInBytesOffset{4 + 1};
   static const int32_t kHeaderSize{kSizeInBytesOffset + 4 + 4 + 8};
 
+  const StreamArena* streamArena_;
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1798]:
```
StreamArena* const streamArena_;
const std::unique_ptr<folly::io::Codec> codec_;
```

File: velox/serializers/PrestoSerializer.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/serializers/PrestoSerializer.cpp:1611]: You put this in ctor initializer list?
```
codec_(xxx)
```

File: velox/serializers/PrestoSerializer.cpp [hunk 18]
```diff
@@ -1595,7 +1607,10 @@ class PrestoVectorSerializer : public VectorSerializer {
       std::shared_ptr<const RowType> rowType,
       int32_t numRows,
       StreamArena* streamArena,
-      bool useLosslessTimestamp) {
+      bool useLosslessTimestamp,
+      dwio::common::CompressionKind compressionKind) {
+    streamArena_ = streamArena;
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1612]: The same for the streamArena_ init

File: velox/dwio/common/Common.cpp [hunk 19]
```diff
@@ -15,11 +15,47 @@
...
+    case folly::io::CodecType::ZLIB:
+      return CompressionKind_ZLIB;
+    case folly::io::CodecType::SNAPPY:
+      return CompressionKind_SNAPPY;
+    case folly::io::CodecType::ZSTD:
+      return CompressionKind_ZSTD;
+    case folly::io::CodecType::LZ4:
+      return CompressionKind_LZ4;
+    default:
+      VELOX_UNSUPPORTED("Not support folly codec type {}", type);
```
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply velox/dwio/common/Common.cpp:55]: If I change to this one, I will get error:
```
this constant expression has type "folly::io::CodecType" instead of the required "int" type
```
Wonder why not switch enum type?

**Comment by jinchengchenghh** [comment]
> Thanks for the update. LGTM. Could you draw a diagram to show the serialized page layout with and without compression in header?

The serialized page layout is same with and without compression in header.
Just change the data layout.

File: velox/serializers/PrestoSerializer.h [hunk 20]
```diff
@@ -15,20 +15,29 @@
...
 #include "velox/vector/VectorStream.h"
 
 namespace facebook::velox::serializer::presto {
 class PrestoVectorSerde : public VectorSerde {
  public:
   // Input options that the serializer recognizes.
   struct PrestoOptions : VectorSerde::Options {
-    explicit PrestoOptions(bool useLosslessTimestamp)
-        : useLosslessTimestamp(useLosslessTimestamp) {}
+    explicit PrestoOptions() {}
```
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply velox/serializers/PrestoSerializer.h:26]: Yes, then we can simplify the function, otherwise, we get nullptr Options most of the case, we should construct the PrestoOptions with its default value.
```
const PrestoVectorSerde::PrestoOptions toPrestoOptions(
    const VectorSerde::Options* options) {
  if (options == nullptr) {
    return PrestoVectorSerde::PrestoOptions();
  }
  return *(static_cast<const PrestoVectorSerde::PrestoOptions*>(options));
}
```

File: velox/dwio/common/Common.cpp (hunk 19 above)
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply velox/dwio/common/Common.cpp:55]: Maybe because `enum class CodecType`

File: velox/serializers/PrestoSerializer.cpp [hunk 21]
```diff
@@ -1757,25 +1854,45 @@ void PrestoVectorSerde::deserialize(
...
+      "Compression kind {} should align with codec marker.",
+      dwio::common::compressionKindToString(
+          dwio::common::codecTypeToCompressionKind(codec->type())));
 
   auto children = &(*result)->children();
   auto childTypes = type->as<TypeKind::ROW>().children();
-  readColumns(source, pool, childTypes, children, useLosslessTimestamp);
+  if (!needCompression(*codec)) {
+    // skip number of columns
+    source->skip(4);
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1880]: Shall we add a check if the number of columns match? We currently see some serialized pages corruption in meta internal test. Thanks!

File: velox/serializers/PrestoSerializer.cpp [hunk 22]
```diff
@@ -1757,25 +1854,45 @@ void PrestoVectorSerde::deserialize(
...
+    auto compressBuf = folly::IOBuf::create(compressedSize);
+    source->readBytes(compressBuf->writableData(), compressedSize);
+    compressBuf->append(compressedSize);
+    auto uncompress = codec->uncompress(compressBuf.get(), uncompressedSize);
+    ByteRange byteRange{
+        uncompress->writableData(), (int32_t)uncompress->length(), 0};
+    ByteStream uncompressedSource;
+    uncompressedSource.resetInput({byteRange});
+    // skip number of columns
+    uncompressedSource.skip(4);
```
Comment (xiaoxmeng) [inline velox/serializers/PrestoSerializer.cpp:1892]: ditto

**Review by xiaoxmeng — APPROVED** [review]
@jinchengchenghh Looks great. Thanks for iterations!

**Comment by xiaoxmeng** [comment]
> > Thanks for the update. LGTM. Could you draw a diagram to show the serialized page layout with and without compression in header?
> 
> The serialized page layout is same with and without compression in header. Just change the data layout. image

Thanks. I meant to add this diagram in the code. But we can do this followup if you'd like to help. Thanks!

File: velox/dwio/common/tests/CommonTest.cpp [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+class CommonTest : public testing::Test {};
+
+TEST_F(CommonTest, compressionKindToCodec) {
+  ASSERT_EQ(
+      folly::io::CodecType::NO_COMPRESSION,
+      compressionKindToCodec(CompressionKind::CompressionKind_NONE)->type());
+  ASSERT_EQ(
+      folly::io::CodecType::LZ4,
+      compressionKindToCodec(CompressionKind::CompressionKind_LZ4)->type());
+  EXPECT_THROW(
```
Comment (mbasmanova) [inline velox/dwio/common/tests/CommonTest.cpp:33]: Use VELOX_ASSERT_THROW

File: velox/serializers/CMakeLists.txt [hunk 24]
```diff
@@ -14,7 +14,7 @@
 add_library(velox_presto_serializer PrestoSerializer.cpp
                                     UnsafeRowSerializer.cpp)
 
-target_link_libraries(velox_presto_serializer velox_vector)
+target_link_libraries(velox_presto_serializer velox_dwio_common velox_vector)
```
Comment (mbasmanova) [inline velox/serializers/CMakeLists.txt:17]: I'm not sure if it is a good idea to add this dependency. What do we need from dwio::common? Can we refactor to extract what we need?

Comment (jinchengchenghh) [reply velox/serializers/CMakeLists.txt:17]: We need this file https://github.com/facebookincubator/velox/blob/main/velox/dwio/common/Common.h, I would suggest to rename this file to Compression.h, and extract it to velox_compression.

File: velox/dwio/common/compression/CMakeLists.txt [hunk 25]
```diff
@@ -12,5 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_library(velox_dwio_common_compression LzoDecompressor.cpp)
+if(${VELOX_BUILD_TESTING})
+  add_subdirectory(tests)
+endif()
+
+add_library(velox_dwio_common_compression Compression.cpp LzoDecompressor.cpp)
```
Comment (mbasmanova) [inline velox/dwio/common/compression/CMakeLists.txt:19]: Does this need to be under dwio? Looks like it will be used for non-DWIO code. Maybe move it somewhere else. It would be nice to do this refactoring in a separate PR.

**Review by xiaoxmeng — COMMENTED** [review]
@jinchengchenghh could you rebase the PR to clear the ci test failure? Thanks!

File: velox/dwio/common/compression/CMakeLists.txt (hunk 25 above)
Comment (jinchengchenghh) [reply velox/dwio/common/compression/CMakeLists.txt:19]: Where would you recommend to put it? Maybe `velox/common/compression` or `velox/compression`? I can help refactor it in a separate  PR.

Comment (mbasmanova) [reply velox/dwio/common/compression/CMakeLists.txt:19]: velox/common/compression sounds good. Thanks.

Comment (xiaoxmeng) [reply velox/dwio/common/compression/CMakeLists.txt:19]: @jinchengchenghh I think you have moved to the wrong location: it should be velox/common/compression/Compression.h but not velox/dwio/common/compression/Compression.h? @mbasmanova

Comment (jinchengchenghh) [reply velox/dwio/common/compression/CMakeLists.txt:19]: I will refactor in a separate PR as this discussion suggested. https://github.com/facebookincubator/velox/pull/5544#discussion_r1265721028

Comment (mbasmanova) [reply velox/dwio/common/compression/CMakeLists.txt:19]: @jinchengchenghh Thank you. Just to clarify, we'll wait for a refactoring PR to proceed.

File: velox/dwio/common/compression/Compression.cpp [hunk 26]
```diff
@@ -0,0 +1,76 @@
...
+    case folly::io::CodecType::ZLIB:
+      return CompressionKind_ZLIB;
+    case folly::io::CodecType::SNAPPY:
+      return CompressionKind_SNAPPY;
+    case folly::io::CodecType::ZSTD:
+      return CompressionKind_ZSTD;
+    case folly::io::CodecType::LZ4:
+      return CompressionKind_LZ4;
+    default:
+      VELOX_UNSUPPORTED("Not support folly codec type {}", type);
```
Comment (tanjialiang) [inline velox/dwio/common/compression/Compression.cpp:55]: type.toString()?

Comment (jinchengchenghh) [reply velox/dwio/common/compression/Compression.cpp:55]: This is a enum class folly::io::CodecType, it does not have toString function

File: velox/serializers/PrestoSerializer.cpp [hunk 27]
```diff
@@ -1580,6 +1595,9 @@ class PrestoVectorSerializer : public VectorSerializer {
     }
   }
 
+  // The SerializedPage layout is:
+  // numRows(4) | codec(1) | uncompressedSize(4) | compressedSize(4) |
+  // checksum(8) | data
```
Comment (tanjialiang) [inline velox/serializers/PrestoSerializer.cpp:1600]: Nice comments. Can we make them triple slash /// as it's public?

File: velox/serializers/PrestoSerializer.cpp [hunk 28]
```diff
@@ -1662,10 +1676,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+    VELOX_CHECK_LE(
+        uncompressedSize,
+        codec_->maxUncompressedLength(),
+        "UncompressedSize exceeds limit");
+    auto compressed = codec_->compress(out.getIOBuf().get());
+    const int32_t compressedSize = compressed->length();
+    writeInt32(output, uncompressedSize);
+    writeInt32(output, compressedSize);
+    const int32_t crcOffset = output->tellp();
+    writeInt64(output, 0); //  // Write zero checksum
```
Comment (tanjialiang) [inline velox/serializers/PrestoSerializer.cpp:1722]: Comments formatting might be off here.

File: velox/serializers/PrestoSerializer.cpp [hunk 29]
```diff
@@ -1662,10 +1676,93 @@ class PrestoVectorSerializer : public VectorSerializer {
...
+    auto listener = dynamic_cast<PrestoOutputStreamListener*>(out->listener());
+    // Reset CRC computation
+    if (listener) {
+      listener->reset();
+    }
+
+    if (!needCompression(*codec_)) {
+      flushUncompressed(numRows, rle, out, listener);
+    } else {
+      flushCompressed(numRows, rle, out, listener);
```
Comment (tanjialiang) [inline velox/serializers/PrestoSerializer.cpp:1756]: If flushUncompressed and flushCompressed are only used internally shall we make them private?

File: velox/dwio/common/compression/Compression.cpp [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+    case CompressionKind_SNAPPY:
+      return "snappy";
+    case CompressionKind_LZO:
+      return "lzo";
+    case CompressionKind_ZSTD:
+      return "zstd";
+    case CompressionKind_LZ4:
+      return "lz4";
+  }
+  return folly::to<std::string>("unknown - ", kind);
```
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply velox/dwio/common/compression/Compression.cpp:74]: This is old code, I just move it to another directory.

File: velox/serializers/PrestoSerializer.h [hunk 31]
```diff
@@ -15,20 +15,29 @@
  */
 #pragma once
 #include "velox/common/base/Crc.h"
+#include "velox/dwio/common/compression/Compression.h"
 #include "velox/vector/VectorStream.h"
 
 namespace facebook::velox::serializer::presto {
 class PrestoVectorSerde : public VectorSerde {
  public:
   // Input options that the serializer recognizes.
```
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply velox/serializers/PrestoSerializer.h:24]: You could open a new PR to do that since it is no relevant change.

**Review by xiaoxmeng — COMMENTED** [review]
@jinchengchenghh file move PR is now committed. Can you rebase? Thanks!

File: velox/serializers/PrestoSerializer.cpp [hunk 32]
```diff
@@ -137,6 +137,18 @@ std::string typeToEncodingName(const TypePtr& type) {
   }
 }
 
+PrestoVectorSerde::PrestoOptions toPrestoOptions(
+    const VectorSerde::Options* options) {
+  if (options == nullptr) {
+    return PrestoVectorSerde::PrestoOptions();
```
[the remark this answers is not in the document]
Comment (xiaoxmeng) [reply velox/serializers/PrestoSerializer.cpp:143]: +1
@jinchengchenghh I know this is due to current code behavior but could you help either fix all the code sites to pass non-null VectorSerde::Options or have a unit test to make sure PrestoVectorSerde::PrestoOptions() returns common::CompressionKind::CompressionKind_NONE.
